### PR TITLE
Add "Self Tests" developer documentation

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by mauigna06** [comment]
Dear @jcfr 

Are the links below supposed to work?

> On-demand execution using:
>         Reload & Test module panel section displayed for scripted modules when the [developer mode](https://github.com/Slicer/Slicer/blob/update-selftests-doc/Docs/developer_guide/modules/user_guide/settings.md#developer) is enabledin the application settings.
>         [Self Tests](https://github.com/Slicer/Slicer/blob/update-selftests-doc/user_guide/modules/selftests.md) module user interface.
> 
> Automatic execution using:
>         CTest in the context of [Nightly tests](https://github.com/Slicer/Slicer/blob/update-selftests-doc/Docs/developer_guide/modules/developer_guide/contributing.md#nightly-tests).

File: Docs/developer_guide/modules/selftests.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## FAQ
+
+Frequently asked questions about how to write and execute tests for Slicer.
+
+### How are the tests executed ?
+
+There are two main mechanisms:
+
+* On-demand execution using:
+  * `Reload & Test` module panel section displayed for scripted modules when the [developer mode](user_guide/settings.md#developer) is enabledin the application settings.
```
Comment (pieper) [inline Docs/developer_guide/modules/selftests.md:12]:
```suggestion
  * `Reload & Test` module panel section displayed for scripted modules when the [developer mode](user_guide/settings.md#developer) is enabled in the application settings.
```

**Review by pieper — DISMISSED** [review]
Small edit but otherwise looks very helpful 👍

**Comment by jcfr** [comment]
Indeed, this link are expected to work. You may be able to preview them browsing the generated documentation.

See https://slicer--6742.org.readthedocs.build/en/6742/developer_guide/modules/selftests.html


For future reference, the preview of the documentation generated with which pull request is available clicking on the highlighted link:

image

**Comment by mauigna06** [comment]
[Related](https://discourse.slicer.org/t/extension-tests-fail-on-cdash-why/26571/15?u=mau_igna_06)

**Comment by jcfr** [comment]
Indeed, the documentation was created to help clarify how tests are executed.

Next step will be to look at the root cause.

In the meantime, do you have specific feedback related to this pull request ?

**Comment by mauigna06** [comment]
> In the meantime, do you have specific feedback related to this pull request ?

Yes, I think it makes test related information easier to find.

Maybe [this link](https://blog.kitware.com/3d-slicer-improves-testing-for-pull-requests-using-docker-and-circleci/) could be interesting for developers

**Comment by jcfr** [comment]
Thanks for pointing this out :pray: 

Since the blog is already linked from these location:
* https://github.com/Slicer/Slicer/blob/main/CONTRIBUTING.md#automatic-testing-of-pull-requests
* https://slicer.readthedocs.io/en/latest/developer_guide/contributing.html#automatic-testing-of-pull-requests

We could probably make this information more discoverable by integrating and actualizing content of the blog by adding a section called `Build infrastructure` in the documentation (see screenshot below).

image


In the meantime, I will move forward with the integration of this pull-request.

**Pull request merged**